### PR TITLE
i18n(es): Update `islands` & `migrate-to-astro`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description (required)

Updated `islands` & `migrate-to-astro` based on this commit 6eb5721f4f3ada61a6ae91b52b9bde8d17024f38

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/es/concepts/islands.mdx [hunk 1]
```diff
@@ -11,11 +11,25 @@ import IslandsDiagram from '~/components/IslandsDiagram.astro';
...
+Astro fue pionero y popularizó una arquitectura de frontend llamada **Islas**. La arquitectura de islas da como resultado un mejor rendimiento del frontend al ayudarte a evitar patrones monolíticos de JavaScript y eliminar automáticamente todo el JavaScript no esencial de la página. Los desarrolladores siguen usando sus componentes y frameworks de UI favoritos con Astro y aún obtienen estos beneficios.
 
-## ¿Qué es una Astro Island?
+## Una breve historia
 
-El término "Astro Island" se refiere a un componente de UI interactivo en una página HTML predominantemente estática. Pueden existir varias islas en una página, y una isla siempre se renderiza en forma aislada. Piensa en ellos como islas en un mar de HTML estático y no interactivo.
+El término "isla de componentes" fue acuñado por primera vez por la arquitecta de frontend de Etsy [Katie Sylor-Miller](https://twitter.com/ksylor) en 2019. Esta idea fue luego expandida y documentada en [este post](https://jasonformat.com/islands-architecture/) por el creador de Preact, Jason Miller, el 11 de agosto de 2020.
```
Comment (Waxer59) [inline src/content/docs/es/concepts/islands.mdx:18]:
```suggestion
El término "isla de componentes" fue acuñado por primera vez por la arquitecta de frontend de Etsy [Katie Sylor-Miller](https://twitter.com/ksylor) en 2019. Esta idea fue luego expandida y documentada en [esta publicación](https://jasonformat.com/islands-architecture/) por el creador de Preact, Jason Miller, el 11 de agosto de 2020.
```

File: src/content/docs/es/concepts/islands.mdx [hunk 2]
```diff
@@ -11,11 +11,25 @@ import IslandsDiagram from '~/components/IslandsDiagram.astro';
...
+
+> La idea general de una arquitectura de "Islas" es engañosamente simple: renderizar páginas HTML en el servidor e inyectar marcadores o espacios reservados alrededor de regiones altamente dinámicas [...] que luego pueden ser "hidratadas" en el cliente en pequeños widgets autocontenidos, reutilizando su HTML inicial renderizado en el servidor.
+> - Jason Miller, Creador de Preact
+Esta técnica se basa en el patrón de arquitectura también conocido como **hidratación parcial** o **selectiva.**
+
+En contraste, la mayoría de los frameworks web basados en JavaScript hidratan y renderizan un sitio web completo como una gran aplicación de JavaScript (también conocida como aplicación de una sola página o SPA). Las SPAs proporcionan simplicidad y potencia, pero sufren de problemas de rendimiento de carga de página debido al uso de JavaScript pesado del lado del cliente.
+
+SPas tienen su lugar, incluso [incrustadas dentro de una página Astro](/es/guides/migrate-to-astro/from-create-react-app/). Pero, las SPAs carecen de la capacidad nativa de hidratación selectiva y estratégica, lo que las convierte en una opción demasiado pesada para la mayoría de los proyectos en la web hoy en día.
```
Comment (Waxer59) [inline src/content/docs/es/concepts/islands.mdx:26]:
```suggestion
Las SPas tienen su lugar, incluso [incrustadas dentro de una página Astro](/es/guides/migrate-to-astro/from-create-react-app/). Pero, las SPAs carecen de la capacidad nativa de hidratación selectiva y estratégica, lo que las convierte en una opción demasiado pesada para la mayoría de los proyectos en la web hoy en día.
```

File: src/content/docs/es/concepts/islands.mdx (hunk 1 above)
Comment (Waxer59) [reply src/content/docs/es/concepts/islands.mdx:18]: No estoy seguro si `component island` deberia ser `isla de componente` o `isla de componentes` 🤔

File: src/content/docs/es/concepts/islands.mdx [hunk 3]
```diff
@@ -11,11 +11,25 @@ import IslandsDiagram from '~/components/IslandsDiagram.astro';
...
 
-## ¿Qué es una Astro Island?
+## Una breve historia
 
-El término "Astro Island" se refiere a un componente de UI interactivo en una página HTML predominantemente estática. Pueden existir varias islas en una página, y una isla siempre se renderiza en forma aislada. Piensa en ellos como islas en un mar de HTML estático y no interactivo.
+El término "isla de componentes" fue acuñado por primera vez por la arquitecta de frontend de Etsy [Katie Sylor-Miller](https://twitter.com/ksylor) en 2019. Esta idea fue luego expandida y documentada en [este post](https://jasonformat.com/islands-architecture/) por el creador de Preact, Jason Miller, el 11 de agosto de 2020.
+
+> La idea general de una arquitectura de "Islas" es engañosamente simple: renderizar páginas HTML en el servidor e inyectar marcadores o espacios reservados alrededor de regiones altamente dinámicas [...] que luego pueden ser "hidratadas" en el cliente en pequeños widgets autocontenidos, reutilizando su HTML inicial renderizado en el servidor.
```
Comment (Waxer59) [inline src/content/docs/es/concepts/islands.mdx:20]:
```suggestion
> La idea general de una arquitectura de "Islas" es engañosamente simple: renderizar páginas HTML en el servidor e inyectar placeholders o slots alrededor de regiones altamente dinámicas [...] que luego pueden ser "hidratadas" en el cliente en pequeños widgets autocontenidos, reutilizando su HTML inicial renderizado en el servidor.
```

**Review by Waxer59 — COMMENTED** [review]
Buena PR 🤠, te dejo unas sugerencias

File: src/content/docs/es/concepts/islands.mdx [hunk 4]
```diff
@@ -11,11 +11,25 @@ import IslandsDiagram from '~/components/IslandsDiagram.astro';
...
+Astro fue pionero y popularizó una arquitectura de frontend llamada **Islas**. La arquitectura de islas da como resultado un mejor rendimiento del frontend al ayudarte a evitar patrones monolíticos de JavaScript y eliminar automáticamente todo el JavaScript no esencial de la página. Los desarrolladores siguen usando sus componentes y frameworks de UI favoritos con Astro y aún obtienen estos beneficios.
 
-## ¿Qué es una Astro Island?
+## Una breve historia
 
-El término "Astro Island" se refiere a un componente de UI interactivo en una página HTML predominantemente estática. Pueden existir varias islas en una página, y una isla siempre se renderiza en forma aislada. Piensa en ellos como islas en un mar de HTML estático y no interactivo.
+El término "isla de componentes" fue acuñado por primera vez por la arquitecta de frontend de Etsy [Katie Sylor-Miller](https://twitter.com/ksylor) en 2019. Esta idea fue luego expandida y documentada en [esta publicación](https://jasonformat.com/islands-architecture/) por el creador de Preact, Jason Miller, el 11 de agosto de 2020.
```
Comment (dreyfus92) [inline src/content/docs/es/concepts/islands.mdx:18]:
```suggestion
El término "isla de componente" fue acuñado por primera vez por la arquitecta de frontend de Etsy [Katie Sylor-Miller](https://twitter.com/ksylor) en 2019. Esta idea fue luego expandida y documentada en [esta publicación](https://jasonformat.com/islands-architecture/) por el creador de Preact, Jason Miller, el 11 de agosto de 2020.
```

**Comment by dreyfus92** [comment]
Muchas gracias por la revisión @Waxer59 😁

**Review by Waxer59 — APPROVED** [review]
LGTM ✅

**Review by sarah11918 — APPROVED** [review]
I see an LGTM... that means MERGE QUEUE!!!  

Thank you @dreyfus92 and @Waxer59 ! These are important content refreshes for 4.0 and it means a lot to us to have them updated! ❤️

**Pull request merged**